src/lib/tabs/tab-group.ts

```diff
@@ -157,19 +155,22 @@ export class MdTabGroup extends _MdTabGroupMixinBase implements AfterContentInit
    * a new selected tab should transition in (from the left or right).
    */
   ngAfterContentChecked(): void {
-    // Clamp the next selected index to the bounds of 0 and the tabs length. Note the `|| 0`, which
-    // ensures that values like NaN can't get through and which would otherwise throw the
-    // component into an infinite loop (since Math.max(NaN, 0) === NaN).
+    // Clamp the next selected index to the boundsof 0 and the tabs length.
+    // Note the `|| 0`, which ensures that values like NaN can't get through
+    // and which would otherwise throw the component into an infinite loop
+    // (since Math.max(NaN, 0) === NaN).
     let indexToSelect = this._indexToSelect =
         Math.min(this._tabs.length - 1, Math.max(this._indexToSelect || 0, 0));
 
     // If there is a change in selected index, emit a change event. Should not trigger if
     // the selected index has not yet been initialized.
     if (this._selectedIndex != indexToSelect && this._selectedIndex != null) {
       this.selectChange.emit(this._createChangeEvent(indexToSelect));
+      // prevent expression changed error
+      this._zone.run(() => this.selectedIndexChange.emit(indexToSelect));
```

Comment:
This makes me wonder if we should bother clamping the tab index, which is the reason why wait for checked content to emit this value. Maybe we should be allowing the dev to input a wrong index like how we do with paginator.

Reply:
Personally I prefer safety checking but if we do not do that on purpose in other places, we should follow the pattern IMO.